test_conformance/api/test_api_min_max.cpp

```diff
@@ -357,13 +364,13 @@ int test_min_max_read_image_args(cl_device_id deviceID, cl_context context,
 
     strcpy(programSrc, sample_read_image_kernel_pattern[0]);
     strcat(programSrc, "read_only image2d_t srcimg0");
-    for (i = 0; i < maxReadImages - 1; i++)
+    for (i = 0; i < maxReadImages - 2; i++)
```

Comment:
I don't understand why this changed from `maxReadImages - 1` to `maxReadImages - 2`.  Seems that this will change the test behavior even for devices that do not approach the SPIR-V function argument limit.  Was this intended?

Same goes for the other changes to loop limits in this PR.

Reply:
This doesn't change original behavior. The test is always running successfully without SPIR-V validation, I am keeping size of function parameter equal to 255. The reason why I am changing code from `maxReadImages - 1` to `maxReadImages - 2` is that there are `__global float *result` and `read_only image2d_t srcimg0` as function parameters, available function parameters are 253. Finally we will get:
`_kernel void sample_test( __global float *result, read_only image2d_t srcimg0, read_only image2d_t srcimg1, ... read_only image2d_t srcimg253 )`

Reply:
ping @bashbaug, thanks very much.

Reply:
Hi, sorry for the delay.

I still think this change needs to be generalized slightly so it will not affect devices that do not approach the SPIR-V function argument limit.

As an experiment, I ran this test on one of our GPU devices with CL_DEVICE_MAX_READ_IMAGE_ARGS = 128 with the OpenCL intercept layer and [DumpProgramSource](https://github.com/intel/opencl-intercept-layer/blob/master/docs/controls.md#dumpprogramsource-bool) enabled, to see how kernels were generated before and after these changes.

Before these changes I see kernels of the form:

```c
__kernel void sample_test( __global float *result, read_only image2d_t srcimg0, ..., read_only image2d_t srcimg127 )
```

So, all 128 images are tested.

After these changes, I see kernels of the form:

```c
__kernel void sample_test( __global float *result, read_only image2d_t srcimg0, ..., read_only image2d_t srcimg126 )
```

So, only 127 images are tested.

Note that this kernel doesn't approach the SPIR-V function argument limit.  Is this intended?  Or, should we be limiting the number of images only when `maxReadImages` is sufficiently large to exceed the SPIR-V function argument limit?

Reply:
Hi @haonanya, want to be sure you saw the comment above - thanks!

Reply:
yes, thanks for your explanation.